### PR TITLE
Admin and User test

### PR DESCRIPTION
**Pull request by ketphan02**
- [x] Admin validation test
- [x] User retrieve test

### CODE REVIEW

File: src/__tests__/pages/api/admin/validate.test.ts
Comment (Opeyem1a) [inline src/__tests__/pages/api/admin/validate.test.ts:116]: What does this do?

File: src/__tests__/pages/api/admin/validate.test.ts
Comment (Opeyem1a) [inline src/__tests__/pages/api/admin/validate.test.ts:103]: Conceptually, what is this testing?

File: src/__tests__/pages/api/admin/validate.test.ts
Comment (Opeyem1a) [inline src/__tests__/pages/api/admin/validate.test.ts:22]: I think this is probably fine for now, but it'd be nice of all these standard tasks like "create admin user with this email" were just functions. Almost like a "test utils" folder. I think it's good in the long run to have the code in tests explicitly be the test itself and not lots of extra set up code that makes files longer than they need to be. Anyway, if this makes sense make a card to do this on Trello and resolve this conversation

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

File: src/__tests__/pages/api/admin/validate.test.ts
Comment (ketphan02) [reply src/__tests__/pages/api/admin/validate.test.ts:116]: oh it not yet done haha, they're fail tests

**Comment by ketphan02** [comment]
Need to wait until #102 to be merged

File: src/__tests__/pages/api/admin/validate.test.ts
Comment (ketphan02) [reply src/__tests__/pages/api/admin/validate.test.ts:22]: I still not get this, please explain in the meeting @Opeyem1a

File: src/__tests__/pages/api/admin/validate.test.ts
Comment (ketphan02) [reply src/__tests__/pages/api/admin/validate.test.ts]: My bad my bad, it shouldn't be like this, I tried a better way

File: src/__tests__/pages/api/auth/reset-password.test.ts
Comment (Opeyem1a) [inline src/__tests__/pages/api/auth/reset-password.test.ts:67]: What do all these `1`s mean?

File: src/__tests__/pages/api/auth/signup.test.ts
Comment (Opeyem1a) [inline src/__tests__/pages/api/auth/signup.test.ts:19]: We don't need the `$connect`/`$disconnect` anymore?

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

File: package.json
Comment (ketphan02) [reply package.json:11]: This make tests run asynchronously
For example, if you are testing creating user and testing creating N multiple users, when you retrive data from the database, it going to be N + 1 users and both tests might fail, `--runInBand` prevents this.

File: src/__tests__/pages/api/auth/reset-password.test.ts
Comment (ketphan02) [reply src/__tests__/pages/api/auth/reset-password.test.ts:67]: Normaly, when sending the requests, you usualy expect it to send 1 time and give you the respond 1 time. The 1's here can spot out if there's any abnormalities.

File: src/__tests__/pages/api/auth/signup.test.ts
Comment (ketphan02) [reply src/__tests__/pages/api/auth/signup.test.ts:19]: Since we have `--runInBand` and we reset the database after every run, we can get rid of it (I'm not sure about this lol, I removed but it still works)

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

**Comment by ketphan02** [comment]
Will have to pull #116 before merging

File: src/__tests__/pages/api/auth/reset-password.test.ts
Comment (Opeyem1a) [inline src/__tests__/pages/api/auth/reset-password.test.ts:198]: What is the argument `2` supposed to be here

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

File: src/__tests__/pages/api/auth/reset-password.test.ts
Comment (ketphan02) [reply src/__tests__/pages/api/auth/reset-password.test.ts:198]: I this test, I call the mock function 2 times, 1 with null para and 1 with undefined para. Therefore, we should expect this mock call function has been called 2 times and returns status 400 both times

**Comment by Opeyem1a** [comment]
Just gotta resolve these conflicts @ketphan02

**Comment by ketphan02** [comment]
@Opeyem1a all done!

**Pull request merged**
